### PR TITLE
b2b customer-confirmation tests

### PR DESCRIPTION
**Pull request by kostiantynlitvinov**
tests for b2b customer-confirmation endpoint

but all of them require manual actions (receiving confirmation key from email) so all of them are commented

## PR Description

TBD

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#     I send a POST request:    /customers/    {"data":{"type":"customers","attributes":{"firstName":"Max","lastName":"Musterman","gender":"Male","salutation":"Mr","email":"maxmusterman@spryker.com","password":"${yves_user_password}","confirmPassword":"${yves_user_password}","acceptedTerms":True}}}
+#     Response status code should be:    201
+#     And Save value to a variable:    [data][id]    userId
+#      # need receive the confirmation key from email
+#     And I send a POST request:    /customer-confirmation   {"data":{"type":"customer-confirmation","attributes":{"registrationKey":"607a17d1c673f461ca40002ea79fddc0"}}}
+#     Response status code should be:    204
+#     And Response reason should be:    No Content
+#     I send a POST request:    /access-tokens    {"data":{"type":"access-tokens","attributes":{"username":"max@spryker.com","password":"${yves_user_password}"}}}
+#     And Response status code should be:    201
+#     And Save value to a variable:    [data][attributes][accessToken]    token
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot:17]: we do not need all this starting from here and till the end of the test except teardown - this test needs to test only confirmation code that works - it does not need to check customer attributes

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot [hunk 2]
```diff
@@ -0,0 +1,115 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+# # can't receive the confirmation from email
+# Customer_confirmation_with_wrong_confirmation_key
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot:7]: done

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+#     Response status code should be:    200
+#     And Response reason should be:    OK
+#     And Response header parameter should be:    Content-Type    ${default_header_content_type}
+#     And Response body has correct self link internal
+#     [Teardown]    Run Keywords    I send a DELETE request:    /customers/${userId}
+#     ...    AND    Response status code should be:    204
+#     ...    AND    Response reason should be:    No Content
+
+# # can't receive the confirmation from email
+# Customer_confirmation_with_empty_confirmation_key
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot:35]: done

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+#     Response status code should be:    200
+#     And Response reason should be:    OK
+#     And Response header parameter should be:    Content-Type    ${default_header_content_type}
+#     And Response body has correct self link internal
+#     [Teardown]    Run Keywords    I send a DELETE request:    /customers/${userId}
+#     ...    AND    Response status code should be:    204
+#     ...    AND    Response reason should be:    No Content
+
+# # can't receive the confirmation from email
+# Customer_confirmation_without_confirmation_key
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot:63]: done

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+#     Response status code should be:    200
+#     And Response reason should be:    OK
+#     And Response header parameter should be:    Content-Type    ${default_header_content_type}
+#     And Response body has correct self link internal
+#     [Teardown]    Run Keywords    I send a DELETE request:    /customers/${userId}
+#     ...    AND    Response status code should be:    204
+#     ...    AND    Response reason should be:    No Content
+
+# # can't receive the confirmation from email
+# Customer_confirmation_with_empty_type
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot:91]: done

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+# # can't receive the confirmation from email
+# Customer_confirmation
+#     I send a POST request:    /customers/    {"data":{"type":"customers","attributes":{"firstName":"Max","lastName":"Musterman","gender":"Male","salutation":"Mr","email":"maxmusterman@spryker.com","password":"${yves_user_password}","confirmPassword":"${yves_user_password}","acceptedTerms":True}}}
+#     Response status code should be:    201
+#     And Save value to a variable:    [data][id]    userId
+#      # need receive the confirmation key from email
+#     And I send a POST request:    /customer-confirmation   {"data":{"type":"customer-confirmation","attributes":{"registrationKey":"607a17d1c673f461ca40002ea79fddc0"}}}
+#     Response status code should be:    204
+#     And Response reason should be:    No Content
+#     I send a POST request:    /access-tokens    {"data":{"type":"access-tokens","attributes":{"username":"max@spryker.com","password":"${yves_user_password}"}}}
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot]: done

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot [hunk 7]
```diff
@@ -0,0 +1,36 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+# # can't receive the confirmation from email
+# Customer_confirmation
+#     I send a POST request:    /customers/    {"data":{"type":"customers","attributes":{"firstName":"Max","lastName":"Musterman","gender":"Male","salutation":"Mr","email":"maxmusterman@spryker.com","password":"${yves_user_password}","confirmPassword":"${yves_user_password}","acceptedTerms":True}}}
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot]: done

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot (hunk 1 above)
Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/positive.robot]: all delete flow moved to teardown

File: tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+#     And Response should return error code:    423
+#     And Response should return error message:    This email confirmation code is invalid or has been already used.
+#     [Teardown]    Run Keywords    I send a POST request:    /access-tokens    {"data":{"type":"access-tokens","attributes":{"username":"${userEmail}","password":"${yves_third_user_password}"}}}
+#     ...    AND    Response status code should be:    201
+#     ...    AND    Save value to a variable:    [data][attributes][accessToken]    token
+#     ...    AND    I set Headers:    Authorization=token
+#     ...    AND    I set Headers:    Authorization=Bearer ${token}
+#     ...    AND    I send a GET request:    /customers/${userId}
+#     ...    AND    Response status code should be:    200
+#     ...    AND    Response reason should be:    OK
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot:53]: GET here is not needed
getting token - please reduce to
I get access token for the customer:    ${email_name}${random}${email_domain}
I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}

Comment (kostiantynlitvinov) [reply tests/api/b2b/glue/customer_endpoints/customer-confirmation/negative.robot]: done

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**